github/admin_stats_test.go

```diff
@@ -0,0 +1,142 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+func TestAdminService_GetAdminStats(t *testing.T) {
+	setup()
+	defer teardown()
+
+	mux.HandleFunc("/enterprise/stats/all", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+
+		fmt.Fprint(w, `
+{
+  "repos": {
+    "total_repos": 212,
+    "root_repos": 194,
+    "fork_repos": 18,
+    "org_repos": 51,
+    "total_pushes": 3082,
+    "total_wikis": 15
+  },
+  "hooks": {
+    "total_hooks": 27,
+    "active_hooks": 23,
+    "inactive_hooks": 4
+  },
+  "pages": {
+    "total_pages": 36
+  },
+  "orgs": {
+    "total_orgs": 33,
+    "disabled_orgs": 0,
+    "total_teams": 60,
+    "total_team_members": 314
+  },
+  "users": {
+    "total_users": 254,
+    "admin_users": 45,
+    "suspended_users": 21
+  },
+  "pulls": {
+    "total_pulls": 86,
+    "merged_pulls": 60,
+    "mergeable_pulls": 21,
+    "unmergeable_pulls": 3
+  },
+  "issues": {
+    "total_issues": 179,
+    "open_issues": 83,
+    "closed_issues": 96
+  },
+  "milestones": {
+    "total_milestones": 7,
+    "open_milestones": 6,
+    "closed_milestones": 1
+  },
+  "gists": {
+    "total_gists": 178,
+    "private_gists": 151,
+    "public_gists": 25
+  },
+  "comments": {
+    "total_commit_comments": 6,
+    "total_gist_comments": 28,
+    "total_issue_comments": 366,
+    "total_pull_request_comments": 30
+  }
+}
+`)
+	})
+
+	stats, _, err := client.Admin.GetAdminStats(context.Background())
+	if err != nil {
+		t.Errorf("AdminService.GetAdminStats returned error: %v", err)
+	}
+
+	want := &AdminStats{
+		Repos: &RepoStats{
+			Int(212),
```

Comment:
I'm not sure we need to test the value of every field, indeed we do not do elsewhere in this package if I recall correctly, as it becomes more challenging to maintain as fields are added.

However, I'm not adverse to it either if you explicitly use the field names here in the test and not rely solely on ordering. That way, a new field can more easily be added, or even if it is not, the unit test would not suddenly break.

Reply:
Absolutely, relying on order is super brittle!